recipe/meta.yaml

```diff
@@ -17,10 +17,10 @@ build:
 
 requirements:
   host:
-    - python >=3.9
+    - python >=3.10
     - pip
   run:
-    - python >=3.9
+    - python >=3.10
     - numpy
     - scipy
     - cartopy
```

Comment:
This section should reflect the latest [`requirements/py312.yml`](https://github.com/SciTools-incubator/iris-esmf-regrid/blob/main/requirements/py312.yml), at least where minimum pins are concerned:

```yaml
# Core dependencies.
  - cartopy>=0.18
  - numpy>=1.14
  - scipy
  - esmpy>=7.0
  - iris
  - cf-units
```